### PR TITLE
linode-cli: 5.26.1 -> 5.45.0

### PR DESCRIPTION
**Pull request by techknowlogick**
## Description of changes

https://github.com/linode/linode-cli/compare/5.26.1...v5.45.0

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/openapi3/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  disabled = pythonOlder "3.8";
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-ohpJBXPYnKaa2ny+WFrbL8pJZCV/bzod9THxKBVFXSw=";
+  };
+
+  nativeBuildInputs = [
+    requests
+    pyyaml
+  ];
```
Comment (natsukium) [inline pkgs/development/python-modules/openapi3/default.nix:23]:
```suggestion
  ];
  
  nativeCheckinputs = [
    pytestCheckHook
  ];
  
  pythonImportsCheck = [
    "openapi3"
  ];
```
Could you please enable checkPhase?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#testing-python-packages-testing-python-packages

File: pkgs/tools/virtualization/linode-cli/default.nix [hunk 2]
```diff
@@ -8,13 +8,16 @@
...
 }:
 
 let
-  sha256 = "0r5by5d6wr5zbsaj211s99qg28nr7wm8iri6jxnksx5b375dah6g";
+  sha256 = "0i73j518jzwp89n9g8qdfp9jnm6syn0sdc1k9hmf0qkk6lazwhi7";
   # specVersion taken from: https://www.linode.com/docs/api/openapi.yaml at `info.version`.
-  specVersion = "4.140.0";
-  specSha256 = "0ay54m4aa8bmmpjc7s66rfzqzk4w25h48b9a665y29g67ybb432g";
+  specVersion = "4.166.0";
+  specSha256 = "0s6kvbdd68bddv0fw72l18nmhci0lac7idw5abch5z7jvm10lk5d";
```
Comment (natsukium) [inline pkgs/tools/virtualization/linode-cli/default.nix:20]: ditto

File: pkgs/tools/virtualization/linode-cli/default.nix [hunk 3]
```diff
@@ -24,12 +27,13 @@ in
 
 buildPythonApplication rec {
   pname = "linode-cli";
-  version = "5.26.1";
+  version = "5.45.0";
+  format = "pyproject";
```
Comment (natsukium) [inline pkgs/tools/virtualization/linode-cli/default.nix:31]:
```suggestion
  pyproject = true;
```

**Review by natsukium — CHANGES_REQUESTED** [review]

**Comment by techknowlogick** [comment]
Thanks for that detailed feedback @natsukium <3
I've made the changes you've requested.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 263074` at d200fd64 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>linode-cli</li>
    <li>python310Packages.openapi3</li>
    <li>python311Packages.openapi3</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**